de.hamstersimulator.objectsfirst.testframework/src/main/java/de/hamstersimulator/objectsfirst/testframework/gamelog/GameLogFactory.java

```diff
@@ -0,0 +1,186 @@
+        this.logEntries.add(logEntry);
+        return logEntry;
+    }
+
+    private LogEntry fromInit(final ObservableInitHamsterCommandSpecification initCommandSpecification) {
+        final HamsterData hamsterData = new HamsterData(
+                initCommandSpecification.getLocation(),
+                this.registerHamster(initCommandSpecification.getHamster()),
+                initCommandSpecification.getNewDirection());
+        final LogEntry logEntry = new InitHamsterLogEntry(hamsterData);
+
+        this.logEntries.add(logEntry);
+        return logEntry;
+    }
+
+    private LogEntry fromMove(final ObservableMoveCommandSpecification moveCommandSpecification) {
+        final LogEntry logEntry = new MoveLogEntry(this.hamsterIdLookup.get(moveCommandSpecification.getHamster()));
+        this.logEntries.add(logEntry);
+        return logEntry;
+    }
+
+    private LogEntry fromTurnLeft(final ObservableTurnLeftCommandSpecification turnCommandSpecification) {
+        final LogEntry logEntry = new TurnLeftLogEntry(this.hamsterIdLookup.get(turnCommandSpecification.getHamster()));
+        this.logEntries.add(logEntry);
+        return logEntry;
+    }
+
+    private LogEntry fromPutGrain(final ObservablePutGrainCommandSpecification putCommandSpecification) {
+        final LogEntry logEntry = new PutGrainLogEntry(this.hamsterIdLookup.get(putCommandSpecification.getHamster()));
+        this.logEntries.add(logEntry);
+        return logEntry;
+    }
+
+    private LogEntry fromPickGrain(final ObservablePickGrainCommandSpecification pickCommandSpecification) {
+        final LogEntry logEntry = new PickGrainLogEntry(this.hamsterIdLookup.get(pickCommandSpecification.getHamster()));
+        this.logEntries.add(logEntry);
+        return logEntry;
+    }
+
+    /**
+     * Creates the GameLog created with this factory
+     *
+     * @return The created GameLog, contains all registered LogEntries
+     */
+    public GameLog toGameLog() {
+        return new GameLog(this.territoryData, this.logEntries);
+    }
+
+    /**
+     * Registers a new Hamster and assigns a unique id for it
+     * This may only be called if this.lock is held
+     *
+     * @param hamster the Hamster to register
+     * @return the assigned id
+     */
+    private int registerHamster(final ObservableHamster hamster) {
+        final int newId = this.idCounter;
+        this.idCounter++;
+        this.hamsterIdLookup.put(hamster, newId);
+        return newId;
+    }
+
+    /**
+     * Takes a Command, and creates a LogEntry based on that command
+     *
+     * @param commandSpecification The CommandSpecification which should be transformed into a log entry
+     * @return The created LogEntry
+     */
+    public LogEntry applyNextCommand(final ObservableCommandSpecification commandSpecification) {
+        this.lock.lock();
+        try {
+            return this.logVisitor.apply(commandSpecification);
```

Comment:
Ist der Vistor hier wirklich nicht reentrant oder ist es eigentlich die Liste, die keine concurrent updates verträgt? Oder soll nebenläufigkeit sequenzalisiert werden?

Reply:
The last one. This should never happen, but better to be safe than sorry